### ISSUE TITLE
Support for being in multiple groups at once?

### ISSUE THREAD

**Issue by smeijer** [mizzao/meteor-partitioner#3]
I couldn't find it in the docs, so I apologize if I missed something and it is already supported.

It should be great if a user could be member of multiple groups at once.

Let's say I need to list all project information that a specific freelancer requires. A freelancer (user) works for multiple `Corporations` and each `Corporation` has a specific `ProjectTeam` the freelancer is member of.

Translated to some document structure, it would be something like:

``` js
   Users: [{ ... }],
   Coorporations: [{
      name: WeDrinkBeers,
      teams: [{ "#1", "#2" }]
      },{
      name: WeLoveLemonade,
      teams: [{ "#A", "#B" }]
   }],
   ProjectTeams: [{
      _id: "#1",
      //...
   },{
      _id: "#2",
      //...
   },{
      _id: "#A",
      //...
   },{
      _id: "#B",
      //...
   }]
```

Now let's say the Freelancer is in the group for Coorporation 1 (WeDrinkBeers), but as he is only part of team 1, he isn't allowed to see information about team 2. An other Freelancer works on both Coorporations, and is member of the ProjectTeams 2, A and B. He wishes to see all his current information at once, but being able to show only the information for one specific ProjectTeam also.

As I see it, each ProjectTeam should be a unique group. But than we need some way to assign a user to multiple groups, and make sure that the current `partition` can be based on multiple groups.

Something like: `Partitioner.setActiveUserGroups(userId, Session.get('showProjectIds'))`

**Comment by thetechnaddict** [comment]
I'm thinking around partitioning by location, avoiding hierarchies - so whilst there may be a group for a City and a group for a neighbourhood of that city they are both groups in their own right - so far a standard partitioning. 

In practice (in the real world)  they need to connect up those groups - sending delegates to each other for instance. 

From your comment above, @mizzao , I am thinking that I can have users in more than one location so long as my interface allows them to switch - that is to say 'leave' one location and 'enter' another.

So for example they could take part in discussions/chats in their city group and then leave that and go and have a chat with their neighbourhood group so long as they don't expect to see both sets of chats at the same time.

Effectively two different logins - 

Have I understood you correctly?

**Comment by mizzao** [comment]
For those who are interested, @kristijanhusak seems to have tried this; looks decent:

> https://github.com/kristijanhusak/meteor-partitioner/tree/multiple-groups

**Comment by kristijanhusak** [comment]
Yes, it's already used on one project, looks like it's working fine for now.

@mizzao do you want me to merge it here?

**Comment by mizzao** [comment]
I'd like to merge it in at some point, but no rush. Good to know that it's there and I'm happy to review and merge when I get a moment.
